feat(game-servers): Improve available gameserver detection

NetroScript opened the pull request:
Sometimes it happens, that a `game server` references a `game`, despite that game not running anymore.

This PR improves the detection of available game servers. Instead if just checking if a game server has a game property, it also loads the game which is attached to the game server to confirm, that the game is still running.

If the game is not running anymore, the game server gets released and returned as available.

Additionally this PR adds a property `endedAt` to the `Game` model. This is used to identify if a cleanup of a game is already finished.

This PR only releases a game server, if the game state is ended or interrupted since at least two minutes.

NetroScript committed:
Add additional check if the game the game server references is running

garrappachc commented on src/game-servers/services/game-servers.service.ts:
```diff
@@ -155,14 +156,41 @@ export class GameServersService implements OnModuleInit {
+    if (availableGameServer !== null) {
+      return availableGameServer;
+    } else {
+      throw new Error.DocumentNotFoundError("No free servers available.");
```
`Error.DocumentNotFoundError` is a part of mongoose's API - please use plain `Error`, or, better yet, a dedicated error class.

NetroScript replied:
I used the mongoose class, because you checked for that in the unit tests

garrappachc commented on src/game-servers/services/game-servers.service.ts:
```diff
@@ -155,14 +156,41 @@ export class GameServersService implements OnModuleInit {
+        if (
+          game.state === GameState.ended ||
+          game.state === GameState.interrupted
+        ) {
```
We also need to make sure the game got cleaned up before taking the gameserver.

NetroScript committed:
Add property endedAt to gameservers 

This property is needed to ensure that the cleanup already happened of a server

garrappachc commented on migrations/1635887923233-add-endedAt-to-game.js:
```diff
@@ -0,0 +1,34 @@
+    .then(() => next());
+};
+
+module.exports.down = (next) => {
```
You can get rid of down() method, we don't use down migrations anywhere in this project.

NetroScript replied:
Ok, removed it

NetroScript committed:
Remove unused .down function for the migration
